utils/cache.js

```diff
@@ -0,0 +1,11 @@
+function removeSuggestionFromCache(cache, suggestion) {
+    for (key in cache) {
+        cache[key] = cache[key].filter(function (element) {
```

Comment:
I don't like functions that mutates object from an argument, this is obscure and it's difficult to track bugs.

Reply:
Also, why are you searching for all keys in the `cache`, isn't the suggestion supposed to be shown only if the query is empty? It should only remove values from `cache['']` if the query must be empty.

Reply:
The local suggestion it is also shown if the prefix is an exact match. So if the user writes exactly what is in local and then removes it we would need to remove it from the entire cache.

Reply:
I altering the function so it return a new instead :wink: